Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/JointLimitRotationManipulators.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/TickBus.h>
+#include <AzToolsFramework/Manipulators/RotationManipulators.h>
+#include <Editor/Plugins/Ragdoll/PhysicsSetupManipulatorCommandCallback.h>
+#include <Editor/Plugins/Ragdoll/PhysicsSetupManipulators.h>
+
+namespace EMotionFX
+{
+    enum class JointLimitFrame : AZ::u8
+    {
+        Parent,
+        Child
+    };
+
+    //! Provides functionality for interactively editing character physics joint limit frame orientations in the Animation Editor Viewport. 
+    class JointLimitRotationManipulators
+        : public PhysicsSetupManipulatorsBase
+        , private AZ::TickBus::Handler
+        , private PhysicsSetupManipulatorRequestBus::Handler
+    {
+    public:
+        JointLimitRotationManipulators(JointLimitFrame jointLimitFrame);
+        ~JointLimitRotationManipulators();
+        void Setup(PhysicsSetupManipulatorData& physicsSetupManipulatorData) override;
+        void Refresh() override;
+        void Teardown() override;
+        void ResetValues() override;
+
+    private:
+        // AZ::TickBus::Handler overrides ...
+        void OnTick(float delta, AZ::ScriptTimePoint timePoint) override;
+
+        // PhysicsSetupManipulatorRequestBus::Handler overrides ...
+        void OnUnderlyingPropertiesChanged() override;
+
+        void OnManipulatorMoved(const AZ::Quaternion& rotation);
+        void BeginEditing();
+        void FinishEditing();
+
+        AzToolsFramework::RotationManipulators m_rotationManipulators;
+        PhysicsSetupManipulatorData m_physicsSetupManipulatorData;
+        JointLimitFrame m_jointLimitFrame = JointLimitFrame::Parent;
+        AZ::Quaternion& GetLocalOrientation();
+        const AZ::Quaternion& GetLocalOrientation() const;
+        MCore::CommandGroup m_commandGroup;
+        AZStd::unique_ptr<PhysicsSetupManipulatorCommandCallback> m_adjustJointLimitCallback;
```

Comment:
uber nit: I think the guideline recommendation is to have all functions first, then all the members.

Reply:
Updated.